pages/flight-search.tsx

```diff
@@ -22,7 +24,37 @@ export default function FlightSearch(props: { flightSelectData: FlightSelectData
     router.push('/flight-select');
   };
 
-  const states = props.flightSelectData.data.states.map((state) => state.name);
+  // const states = props.flightSelectData.data.states.map((state) => state.name);
+
+  const fetcher = (url: URL, payload?: string) => {
+    const options = {
+      method: payload ? 'POST' : 'GET',
+      ...(payload && { body: payload }),
+      headers: {
+        accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    };
+
+    return fetch(url, options).then((res) => res.json());
+  };
+
+  const bodyContent = { country: 'United States' };
+
+  const { data, error } = useSWR<FlightSelectData>(
+    ['https://countriesnow.space/api/v0.1/countries/states', JSON.stringify(bodyContent)],
+    fetcher,
+  );
+
+  useEffect(() => {
+    if (data) {
+      setTimeout(() => {
+        setLoading(!data && !error);
+        const statesData = (data as FlightSelectData).data.states.map((state: { name: string }) => state.name);
+        setStates(statesData);
```

Comment:
did this end up being responsible for the dom reload?

Reply:
Not entirely sure if using a state variable was what was necessary because the example from the swr docs didn't use state. We could try a simple test where we update a local variable and see if it re-renders the fragment.